src/unstract/sdk/metrics_mixin.py

```diff
@@ -0,0 +1,51 @@
+import os
+import time
+import uuid
+
+from redis import StrictRedis
+
+
+class MetricsMixin:
+    def __init__(self, run_id):
+        """Initialize the MetricsMixin class.
+
+        Args:
+            run_id (str): Unique identifier for the run.
+        """
+        self.run_id = run_id
```

Comment:
@Deepak-Kesavan can you confirm if this `run_id` will be unique for an entire tool run / file being executed? Where is it generated from? From [here](https://github.com/Zipstack/unstract/blob/3c491e51a07a1053f73687e2c05dfa72c9c1ebc5/backend/workflow_manager/workflow_v2/workflow_helper.py#L246)?

Reply:
Discussed this offline - we need to ensure that the `run_id` mentioned in the link above needs to be passed down to the tools and used while reporting metrics

Reply:
Yes @chandrasekharan-zipstack . I believe it is not being done like that currently. Need to handle it in unstract side.